stumpy/Mmotifs.py

```diff
@@ -0,0 +1,297 @@
+
+        # Choose dimension k using MDL
+        mdl, subspaces = stumpy.mdl(T, m, candidate_idx, nn_idx)
+        k = np.argmin(mdl)
+
+        # Calculate default cutoff value
+        if cutoff is None:
+            P_copy = P.copy().astype(np.float64)
+            P_copy[np.isinf(P_copy)] = np.nan
+            P_copy = P_copy[k, :]
+            cutoff_var = np.nanmax(
+                [np.nanmean(P_copy - 2.0 * np.nanstd(P_copy)), np.nanmin(P_copy)]
+            )
+            cutoff = None
+        else:
+            if isinstance(cutoff, np.ndarray) or isinstance(cutoff, list):
+                cutoff_var = cutoff[k]
+            else:
+                cutoff_var = cutoff
+
+        # Get k-dimensional motif
+        motif_idx = candidate_idx[k]
+        motif_value = P[k, motif_idx]
+        if motif_value > cutoff_var or not np.isfinite(motif_value):
+            break
+
+        if visualize_mdl:
+            # Visualize MDL results:
+            plt.plot(np.arange(len(mdl)), mdl, linewidth="4")
+            plt.title("MDL results", fontsize="30")
+            plt.xlabel("k (zero-based)", fontsize="20")
+            plt.ylabel("Bit Size", fontsize="20")
+            plt.show()
+
+        # Compute subspace and find related dimensions
+        subspace.append(subspaces[k])
+        sub_dims = T[subspaces[k]]
+
+        # Stop iteration if max_distance is a constant and the k-dim.
+        # matrix profile value is larger than the maximum distance.
+        if isinstance(max_distance, float) and motif_value > max_distance:
+            break
+
+        # Get multidimensional Distance Profile to find the max_matches
+        # nearest neighbors
+        T_sub, mean_T, sigma_T = core.preprocess(sub_dims, m)
+
+        Q = sub_dims[:, motif_idx : motif_idx + m]
+        query_matches = Mmatch(
+            Q=Q,
+            T_sub=sub_dims,
+            mean_T=mean_T,
+            sigma_T=sigma_T,
+            max_matches=max_matches,
+            max_distance=max_distance,
+            atol=atol,
+        )
+        if len(query_matches) > min_neighbors:
+            motif_matches_distances.append(query_matches[:, 0])
+            motif_matches_indices.append(query_matches[:, 1])
+
+        # Set exclusion zones and find new candidate_idx an nn_idx for
+        # the next motif
+        for idx in query_matches[:, 1]:
+            core.apply_exclusion_zone(P, idx, excl_zone, np.inf)
+        candidate_idx = np.argsort(P, axis=1)[:, 0]
+        nn_idx = I[np.arange(len(candidate_idx)), candidate_idx]
+
+    motif_matches_distances = core._jagged_list_to_array(
+        motif_matches_distances, fill_value=np.nan, dtype=np.float64
+    )
+    motif_matches_indices = core._jagged_list_to_array(
+        motif_matches_indices, fill_value=-1, dtype=np.int64
+    )
+
+    return (
+        motif_matches_distances,
+        motif_matches_indices,
+        np.array(subspace, dtype=object),
+    )
+
+
+def Mmatch(

```

Comment:
This entire function looks identical to `stumpy.match`. Let's not repeat code and, instead, `from . import match` have `mmotif` call `match` instead